rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2319,8 +2319,14 @@ namespace rsx
 		vm::_ref<atomic_t<CellGcmReportData>>(sink).store({ timestamp(), value, 0});
 	}
 
-	void thread::sync()
+	void thread::sync(bool fifo)
 	{
+		if (fifo)
```

Comment:
The fifo path is unique and does not interact with the rest of the code, making this seem like embedded code that does not belong. Just declare a proper method for it, sync_fifo or flush_fifo. I prefer flush_fifo.